plutus-core/src/Language/PlutusCore/Parsable.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE DefaultSignatures    #-}
+{-# LANGUAGE FlexibleInstances    #-}
+{-# LANGUAGE OverloadedStrings    #-}
+{-# LANGUAGE TypeApplications     #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+{- | A typeclass which provides a `parseConstant` method to convert Text strings
+into objects of the appropriate type. This allows one to define parsers for
+literal constants belonging to extensible built-in types without having to
+modify the main Plutus Core parser. We expect parseConstant . prettyConst to be
+the identity function. -}
+
+module Language.PlutusCore.Parsable
+    (
+     Parsable(..)
+    )
+where
+
+import           PlutusPrelude
+
+import           Data.Bits            (shiftL, (.|.))
+import           Data.ByteString.Lazy (ByteString)
+import qualified Data.ByteString.Lazy as BSL (pack)
+import           Data.Char            (ord)
+import qualified Data.Text            as T
+import           Text.Read            (readMaybe)
+
+
+class Parsable a
+  where
+  parseConstant :: T.Text -> Maybe a
+  -- ^ Return Nothing if the string is invalid, otherwise the corresponding value of type a.
+  default parseConstant :: Read a => T.Text -> Maybe a
+  parseConstant = readMaybe @ a . T.unpack
```

Comment:
Do you need the `@ a`?

Reply:
Apparently not.  I'm sure I needed it at some point, but not any more.